master/internal/resourcemanagers/resource_pool.go

```diff
@@ -381,36 +381,136 @@ func (rp *ResourcePool) receiveAgentMsg(ctx *actor.Context) error {
 	return nil
 }
 
-func (rp *ResourcePool) moveJob(msg job.MoveJob) error {
-	// REMOVEME
-	fmt.Println(rp.config.PoolName, "moveJob: ", msg)
-	rp.queuePositions[job.TailAnchor] = initalizeQueuePosition(time.Now())
-	newPos, rebalance, err := computeNewJobPos(msg, rp.queuePositions)
-	if err != nil {
-		return err
+func (rp *ResourcePool) moveJob(
+	ctx *actor.Context, jobID model.JobID, anchorID model.JobID, aheadOf bool,
+) error {
+	if rp.config.Scheduler.GetType() != priorityScheduling {
+		return nil
+	}
+
+	if anchorID == "" || jobID == "" || anchorID == jobID {
+		return nil
+	}
+
+	if _, ok := rp.queuePositions[jobID]; !ok {
+		return nil
+	}
+
+	groupAddr, ok := rp.IDToGroupActor[jobID]
+	if !ok {
+		return job.ErrJobNotFound(jobID)
 	}
-	rp.queuePositions[msg.ID] = newPos
-	// condiontally check if rebalancing is needed.
-	if rebalance {
-		adjustPriorities(&rp.queuePositions)
+
+	if _, ok := rp.queuePositions[anchorID]; !ok {
+		return job.ErrJobNotFound(anchorID)
+	}
+
+	prioChange, secondAnchor, anchorPriority := rp.findAnchor(jobID, anchorID, aheadOf)
+
+	check.Panic(check.True(secondAnchor != ""))
+
+	if secondAnchor == jobID {
+		return nil
+	}
+
+	if prioChange {
+		resp := ctx.Self().System().AskAt(job.JobsActorAddr, job.SetJobPriority{ID: jobID, Priority: anchorPriority})
```

Comment:
bit strange this goes through the jobs actor rather than just directly to the job, but i get why, but i also think it could not

Reply:
hmm Ill look into it, but yeah the jobs actor is supposed to act almost like an RM for jobs

Reply:
this is more just commentary, for now. but i think the rm could just ask the job directly, personally

Reply:
like, less complexity is always better, and i think the jobs actor could be basically just a thin proxy to help apis. if you think of the tell/ask graph, this is a totally new line whereas we could just use old paths.

Reply:
eh, i think this is blocking for me. why ask job.JobsActorAddr SetJobPriority so it can ask jobActor SetGroupPriority rather than just asking groupAddr SetGroupPriority?

Reply:
and then SetJobPriority can be removed entirely